docs/about-stacks.md

```diff
@@ -24,47 +24,44 @@ A terraform stack is a runnable terraform module that operates on a subset of
 the infrastructure's resources.
 ```
 
-By _runnable_ it means it has everything needed to call
-`terraform init/plan/apply` . For example, a module used by other modules, ie,
-don't have the `provider` explicitly set, is not runnable hence it's
-**not a stack**.
+For a module to be considered a stack, it must be complete and self-contained, 
+meaning that it can run all the necessary Terraform commands (`init/plan/apply`) 
+without any external dependencies.
```

Comment:
Shall we make more clear that provider and Terraform configuration is required? I would make this way more explicit, especially to distinguish between modules and stacks. wdyt @mariux ?

Reply:
A stack is not limited to terrform only.. it can be a kubernetes stack, pure Terraform, Terrgraunt or anything..

In Terraform context (most common) a stack is what hashicorop call a root module.. it has a single state by default but when using workspaces can have multiple states (one per workspace)..

Terraform defaults to a local backen if none is defined.. Terraform also just downloads the latest version of all used providers it is aware of, so most configurations work without defining any of those.. 

Best practice demand to add such cibfiguratio anyway and terraform will warn if it is not defined but not fail.. 

Terramate integrates with Terraform stacks the beat atm as change detection is aware of any dependendciesike local modules.. 

Terraform itself does not fully distinguish between stacks (root modules) and modules itself.. with the newly released zero config modules it actually breaks with everything that was before a good practice.. ;)

Tereamte does not demand anything but having a stack config block defined in any Terramate config file within the stack.. 

using tags you can define intelligent filtering to even run multiple types stacks in a single repository.. or having a single stack being a kubernetes and Terraform stack at the same time.. 

We can define best practices though for each stack type.. e.g. a Terramate Terraform Stack should have provider and backend config defined and include versioning constraints for providers and Terraform itself.. 

Bottom line is, we are not limited to Terraform.. and do not demand anything but an (Empty) stack block as a minimal setup.. at least in Terramte CLI this is enough to get it going.. ;) 

We will add better onbaording in a soon to come Terramte release for existing Terraform configurations.. Also provide easy ways to add a inquiry stack id to existing stacks that miss one..